bundles/org.openhab.binding.orbitbhyve/src/main/java/org/openhab/binding/orbitbhyve/internal/handler/OrbitBhyveBridgeHandler.java

```diff
@@ -0,0 +1,584 @@
+        if (event != null) {
+            processEvent(event);
+        }
+    }
+
+    private void processEvent(OrbitBhyveSocketEvent event) {
+        switch (event.getEvent()) {
+            case "watering_in_progress_notification":
+                disableZones(event.getDeviceId());
+                Channel channel = getThingChannel(event.getDeviceId(), event.getStation());
+                if (channel != null) {
+                    logger.debug("Watering zone: {}", event.getStation());
+                    updateState(channel.getUID(), OnOffType.ON);
+                    String program = event.getProgram().getAsString();
+                    if (!program.isEmpty() && !"manual".equals(program)) {
+                        channel = getThingChannel(event.getDeviceId(), "program_" + program);
+                        if (channel != null) {
+                            updateState(channel.getUID(), OnOffType.ON);
+                        }
+                    }
+                }
+                break;
+            case "watering_complete":
+                logger.debug("Watering complete");
+                disableZones(event.getDeviceId());
+                disablePrograms(event.getDeviceId());
+                updateDeviceStatus(event.getDeviceId());
+                break;
+            case "change_mode":
+                logger.debug("Updating mode to: {}", event.getMode());
+                Channel ch = getThingChannel(event.getDeviceId(), CHANNEL_MODE);
+                if (ch != null) {
+                    updateState(ch.getUID(), new StringType(event.getMode()));
+                }
+                ch = getThingChannel(event.getDeviceId(), CHANNEL_CONTROL);
+                if (ch != null) {
+                    updateState(ch.getUID(), "off".equals(event.getMode()) ? OnOffType.OFF : OnOffType.ON);
+                }
+                deferredStatusUpdate(event.getDeviceId());
+                break;
+            case "rain_delay":
+                deferredStatusUpdate(event.getDeviceId());
+                break;
+            case "skip_active_station":
+                disableZones(event.getDeviceId());
+                break;
+            case "program_changed":
+                OrbitBhyveProgram program = gson.fromJson(event.getProgram(), OrbitBhyveProgram.class);
+                if (program != null) {
+                    updateDeviceProgramStatus(program);
+                    deferredStatusUpdate(program.getDeviceId());
+                }
+                break;
+            default:
+                logger.debug("Received event: {}", event.getEvent());
+        }
+    }
+
+    private void deferredStatusUpdate(String deviceId) {
+        ScheduledFuture<?> localFuture = statusFuture;
+        if (localFuture != null) {
+            localFuture.cancel(true);
+        }
+        statusFuture = scheduler.schedule(() -> updateDeviceStatus(deviceId), 3, TimeUnit.SECONDS);
```

Comment:
This won't work if the user sends two commands within less than 3 seconds, as `statusFuture` will be overwritten, then.

Are you sure the update is needed at all? Actually, openHAB predicts the new state automatically when a command is sent to a channel. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#auto-update-policies

Reply:
I have changed the behaviour so it call the device status immediately and ping method does status reconciliation, so any delayed status updates are taken into account.

Reply:
Actually, you don't need to request the new state from the device as the framework predicts the new state immediately and verifies it during the next polling. But it's ok to keep it for now.